controller/konnect/reconciler_generic.go

```diff
@@ -1042,6 +1051,11 @@ func handleControlPlaneRef[T constraints.SupportedKonnectEntityType, TEnt constr
 			Name:      cpRef.KonnectNamespacedRef.Name,
 			Namespace: ent.GetNamespace(),
 		}
+		// Set namespace of control plane when it is non-empty. Only applyies for cluster scoped resources (KongVault).
+		// REVIEW: add type check here?
```

Comment:
Since this is enforced at the CRD level I believe we can skip the type check here. WDYT?

Reply:
Done.